library/src/math/cancel.rs

```diff
@@ -0,0 +1,171 @@
+    pub invert: bool,
+
+    /// If two opposing cancel lines should be drawn, forming a cross over the element.
+    /// Overrides `invert`.
+    ///
+    /// Defaults to `{false}`.
+    #[default(false)]
+    pub cross: bool,
+
+    /// Rotate the cancel line by a certain angle. See the
+    /// [line's documentation]($func/line.angle) for more details.
+    #[default(Angle::zero())]
+    pub rotate: Angle,
+
+    /// How to stroke the cancel line. See the
+    /// [line's documentation]($func/line.stroke) for more details.
+    #[resolve]
+    #[fold]
+    pub stroke: PartialStroke,
+}
+
+impl LayoutMath for CancelElem {
+    fn layout_math(&self, ctx: &mut MathContext) -> SourceResult<()> {
+        let mut body = ctx.layout_frame(&self.body())?;
+
+        let size = body.size();
+        let (width, height) = (size.x, size.y);
+
+        let length = self
+            .length(ctx.styles()) // empirically pleasant default
+            .unwrap_or(Rel::new(Ratio::one(), Abs::pt(3.0).into()))
+            .resolve(ctx.styles());
+
+        // default stroke has 0.5pt for better visuals
+        let stroke = self.stroke(ctx.styles()).unwrap_or(Stroke {
+            paint: TextElem::fill_in(ctx.styles()),
+            thickness: Abs::pt(0.5),
+            ..Default::default()
+        });
+
+        let invert = self.invert(ctx.styles());
+        let cross = self.cross(ctx.styles());
+        let angle = self.rotate(ctx.styles());
+
+        let invert_first_line = !cross && invert;
+
+        let first_line = draw_cancel_line(
+            length,
+            stroke.clone(),
+            invert_first_line,
+            angle,
+            size,
+            self.span(),
+        );
+
+        // the origin of our line is the very middle of the element
+        body.push_frame(Point::new(width / 2.0, height / 2.0), first_line);
+
+        if cross {
+            // draw the second line
+            let second_line = draw_cancel_line(
+                length,
+                stroke,
+                /*invert:*/ true,
+                angle,
+                size,
+                self.span(),
+            );
+
+            body.push_frame(Point::new(width / 2.0, height / 2.0), second_line);
+        }
+
+        ctx.push(FrameFragment::new(ctx, body));
+
+        Ok(())
+    }
+}
+
+/// Draws a cancel line.
+fn draw_cancel_line(
+    length: Rel<Abs>,
+    stroke: Stroke,
+    invert: bool,
+    angle: Angle,
+    body_size: Size,
+    span: Span,
+) -> Frame {
+    let (width, height) = (body_size.x, body_size.y);
+    let (width_pt, height_pt) = (width.to_pt(), height.to_pt());
+
+    //              /|
+    // diagonal_pt / |  height_pt
+    //            /  |
+    //           /   |
+    //           -----
+    //          width_pt
+    let diagonal_pt = width_pt.hypot(height_pt);
+    let diagonal = Abs::pt(diagonal_pt);
+
+    let length = length.relative_to(diagonal);
+
+    let mid_x = width / 2.0;
+    let mid_y = height / 2.0;
```

Comment:
a few computations could benefit from being done directly in composite types like:
`let mid = body_size.to_point() / 2.0`.

Reply:
I did this directly with `body_size` instead of `body_size.to_point()`, as only `mid`'s coordinates are used later (it's not used as a point); this should be fine, right?

Reply:
yes